src/types/types.ts

```diff
@@ -805,6 +805,165 @@ export interface GetFloorPriceResponse {
   readonly looksRare: FloorPriceMarketplace | FloorPriceError;
 }
 
+/**
+ * Optional parameters object for the {@link getNftSales} endpoint.
+ *
+ * This interface is used to filter the NFT sales data.
+ *
+ * @public
+ */
+export interface GetNftSales {
+  /** The block number to start fetching NFT sales data from. */
+  fromBlock?: number | 'latest';
+
+  /** The block number limit to fetch NFT sales data from. */
+  toBlock?: number | 'latest';
+
+  /** Whether to return the results in ascending or descending order by block number. */
+  order?: SortingOrder;
+
+  /** The NFT marketplace to filter sales by. */
+  marketplace?: NftSaleMarketplace;
+
+  /** The address of the NFT buyer to filter sales by. */
+  buyerAddress?: string;
+
+  /** The address of the NFT seller to filter sales by. */
+  sellerAddress?: string;
+
+  /** Filter by whether the buyer or seller was the taker in the NFT trade. */
+  taker?: NftTakerType;
+
+  /** The maximum number of NFT sales to return. */
+  limit?: number;
+
+  /** Key for pagination to use to fetch results from the next page if available. */
+  pageKey?: string;
+}
+
+/**
+ * Alternative optional parameters object for the {@link getNftSales} endpoint
+ * that allows filtering results by contractAddress.
+ *
+ * This interface is used to filter the NFT sales data.
+ *
+ * @public
+ */
+export interface GetNftSalesByContractAddress extends GetNftSales {
+  /** The contract address of a NFT collection to filter sales by. */
+  contractAddress: string;
+
+  /** The token ID of an NFT within the specified contractAddress to filter sales by. */
+  tokenId?: BigNumberish;
+}
+
+/**
+ * The response for the {@link NftNamespace.getNftSales} method.
+ *
+ * @public
+ */
+export interface GetNftSalesResponse {
+  /** The page key to use to fetch the next page if more results are available. */
+  pageKey?: string;
+
+  /** List of NFT sales that match the query */
+  nftSales: NftSale[];
+}
+
+/** A single NFT sale data. */
+export interface NftSale {
+  /** The marketplace the sale took place on. */
+  marketplace: NftSaleMarketplace;
+
+  /** The NFT contract address. */
+  contractAddress: string;
+
+  /** The decimal token ID of the NFT being sold. */
+  tokenId: string;
+
+  /** The number of tokens sold in the sale as a decimal integer string. */
+  quantity: string;
+
+  /** The address of the buyer in the NFT sale. */
+  buyerAddress: string;
+
+  /** The address of the seller in the NFT sale. */
+  sellerAddress: string;
+
+  /** Whether the price taker in the trade was the buyer or the seller. */
+  taker: NftTakerType;
+
+  /** The payment from buyer to the seller. */
+  sellerFee?: NftSaleFeeData;
```

Comment:
Please remove optional on the property, since sellerFee is always included in the response.